src/components/embedded/EmbeddedMessages.tsx

```diff
@@ -248,11 +256,23 @@ const MessagesVirtualizedList = observer((props: Props) => {
 		debouncedScrollHandler(event, event.deltaY < 0 ? 'next' : 'previous');
 	};
 
+	const itemRenderer = React.useCallback(
```

Comment:
i think its still better to accept itemRenderer as a prop.
you can do it by creating another another function like so

```
itemContent={(index, message) => {
  return itemRenderer(message, displayType)
}}
```

also we can omit index parameter in itemRenderer since it isnt very usefull

Reply:
index is required by itemContent prop of Virtuoso, so I think we can't delete it